plugins/scripts/upload/trns_transform_Genbank_Genome_to_KBaseGenomeAnnotations_GenomeAnnotation.py

```diff
@@ -1677,23 +1683,86 @@ def upload_genome(shock_service_url=None,
             feature_container['type']= feature_type
             feature_container['features'] = features_type_containers_dict[feature_type]
             feature_container['assembly_ref'] = assembly_reference
-            counts_map[feature_type] = len(features_type_containers_dict[feature_type])
-            #Provencance has a 1 MB limit.  We may want to add more like the accessions, but to be safe for now not doing that.
+
+            #Provenance has a 1 MB limit.  We may want to add more like the accessions, but to be safe for now not doing that.
             #provenance_description = "features from upload from %s includes accession(s) : " % (source_name,",".join(locus_name_order))
             feature_container_provenance = [{"script": __file__, "script_ver": "0.1", "description": "features from upload from %s" % (source_name)}]
-
             logger.info("Attempting save of Feature Container %s" % (feature_container_object_name)) 
 
+            do_feature_container_save = True
+
+            container_byte_estimate = len(simplejson.dumps(feature_container))  
+
+            container_feature_lengths = OrderedDict()
+            features_with_sequences_removed_list = list()
+            #1073741824 bytes is a GB.  This gives ~7%cushion for potential unicode 16 characters
+            if container_byte_estimate > 1000000000 : 
+                #Create an ordered dict of features and lengths
+                temp_dict = dict()
+                for feature in feature_container['features']:
+                    temp_dict[feature] = feature_container['features'][feature]['dna_sequence_length']
+                container_feature_lengths = OrderedDict(sorted(temp_dict.items(), key=lambda t: t[1]))
+            while container_byte_estimate > 1000000000 :
+                remove_count = 10
+                container_length = len(container_feature_lengths)
+                if container_length < 10 :
+                    remove_count = container_length
+                if container_length == 0:
+                    #IF CORE TYPE THROW AN ERROR
+                    if feature_container['type'] in ['gene','CDS','mRNA']:
```

Comment:
If the container has no more features, how is it bigger than 1GB? What other things are consuming space? And, if there are other things that large, could we check them first instead of removing all the features and then seeing that it's too big anyways?